src/sqlfluff/core/linter/linter.py

```diff
@@ -494,112 +495,142 @@ def lint_fix_parsed(
+                        ignore_mask=ignore_buff,
+                        fname=fname,
+                    )
+                    all_linting_errors += linting_errors
+
+                    if fix and fixes:
+                        linter_logger.info(f"Applying Fixes [{crawler.code}]: {fixes}")
+                        # Do some sanity checks on the fixes before applying.
+                        anchor_info = BaseSegment.compute_anchor_edit_info(fixes)
+                        if any(
+                            not info.is_valid for info in anchor_info.values()
+                        ):  # pragma: no cover
+                            message = (
+                                f"Rule {crawler.code} returned conflicting "
+                                "fixes with the same anchor. This is only "
+                                "supported for create_before+create_after, so "
+                                "the fixes will not be applied. {fixes!r}"
+                            )
+                            cls._report_conflicting_fixes_same_anchor(message)
+                            for lint_result in linting_errors:
+                                lint_result.fixes = []
+                        elif fixes == last_fixes:  # pragma: no cover
+                            # If we generate the same fixes two times in a row,
+                            # that means we're in a loop, and we want to stop.
+                            # (Fixes should address issues, hence different
+                            # and/or fewer fixes next time.)
+                            cls._warn_unfixable(crawler.code)
+                        else:
+                            # This is the happy path. We have fixes, now we want to
+                            # apply them.
+                            last_fixes = fixes
+                            new_tree, _ = tree.apply_fixes(
+                                config.get("dialect_obj"), crawler.code, anchor_info
+                            )
+                            # Check for infinite loops
+                            if new_tree.raw not in previous_versions:
+                                # We've not seen this version of the file so
+                                # far. Continue.
+                                tree = new_tree
+                                previous_versions.add(tree.raw)
+                                changed = True
+                                continue
+                            else:
+                                # Applying these fixes took us back to a state
+                                # which we've seen before. We're in a loop, so
+                                # we want to stop.
+                                cls._warn_unfixable(crawler.code)
+
+                if phase == "main" and loop == 0:
+                    # Keep track of initial errors for reporting.
+                    initial_linting_errors = all_linting_errors.copy()
```

Comment:
This is where we capture `initial_linting_errors`, which is what the function returns (and what the user sees in the list of violations). See above -- this pass runs all rules (whether main or post, whether they are "fix compatible" or not. So the user sees a list of initial errors, but they don't see any linting errors introduced by fixes. This is the past behavior and remains true with these changes.

I'll see about some code comments and/or small code changes to make this clearer.